core/extension/gdextension.cpp

```diff
@@ -491,6 +491,13 @@ void GDExtension::_register_extension_class_internal(GDExtensionClassLibraryPtr
 	}
 #endif
 
+	if (!extension->gdextension.gdtype) {
+		// When hot reloading, the gdtype already exists.
+		// For now, we don't need to do anything else to the GDType,
+		// since none of the hot reloadable attributes are in GDType yet.
+		extension->gdextension.create_gdtype();
+	}
```

Comment:
`GDType` does keep track of something that can change during reload: the list of parent classes.

On reload, the parent class within Godot can't change, but the parent class within the extension can.

For example, if `MyExtensionClass` extends `Node`, after reload it can't switch to extending `Control`. But it can switch to extending `MyOtherExtensionClass` which extends `Node`. So, we'd need to update `GDType::get_name_heirarchy()` somehow